src/main/java/core/basesyntax/service/date/ReportGeneratorImpl.java

```diff
@@ -0,0 +1,23 @@
+package core.basesyntax.service.date;
+
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class ReportGeneratorImpl implements ReportGenerator {
+    private Map<String, Integer> storage;
```

Comment:
Following the encapsulation principle, the `storage` field should be private and final, and it should be initialized only in the constructor. This would ensure that the `storage` reference is not changed after it is initialized.